### PR TITLE
Service datasource

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Acceptance testing result
image

### CODE REVIEW

File: powerflex/provider/service_datasource_schema.go [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/setvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// TemplateDataSourceSchema defines the schema for template datasource
```
Comment (shenda1) [inline powerflex/provider/service_datasource_schema.go:28]: service

File: powerflex/provider/service_datasource_schema.go [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+			},
+		},
+		"service_names": schema.SetAttribute{
+			Description:         "List of service names",
+			MarkdownDescription: "List of service names",
+			Optional:            true,
+			ElementType:         types.StringType,
+			Validators: []validator.Set{
+				setvalidator.SizeAtLeast(1),
+				setvalidator.ConflictsWith(
```
Comment (shenda1) [inline powerflex/provider/service_datasource_schema.go:57]: I think this check is redundant

File: powerflex/provider/service_datasource_schema.go [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+			Description:         "Retry",
+			Computed:            true,
+		},
+		"teardown": schema.BoolAttribute{
+			MarkdownDescription: "teardown",
+			Description:         "teardown",
+			Computed:            true,
+		},
+		"teardown_after_cancel": schema.BoolAttribute{
+			MarkdownDescription: "teardown_after_cancel",
```
Comment (shenda1) [inline powerflex/provider/service_datasource_schema.go:106]: Can we correct the description? Looks like attribute name is copied for all the attributes.

File: powerflex/models/service.go [hunk 4]
```diff
@@ -34,3 +34,81 @@ type ServiceResourceModel struct {
 	TemplateName          types.String `tfsdk:"template_name"`
 	DeploymentTimeout     types.Int64  `tfsdk:"deployment_timeout"`
 }
+
+// TemplateDataSourceModel maps the struct to Template data source schema
```
Comment (shenda1) [inline powerflex/models/service.go:38]: Service

**Review by shenda1 — COMMENTED** [review]
Please add coverage report

File: powerflex/provider/service_datasource.go [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+			if err != nil {
+				resp.Diagnostics.AddError(
+					fmt.Sprintf("Error in getting service details using id %v", serviceID), err.Error(),
+				)
+				return
+			}
+			serviceModel = append(serviceModel, helper.GetDataSourceServiceState(*serviceDetails))
+		}
+	} else if !state.ServiceNames.IsNull() {
+		// Fetch Service details if IPs are provided
```
Comment (shenda1) [inline powerflex/provider/service_datasource.go:105]: IPs -> Names?

**Comment by AnikaAgiwal2711** [comment]
image

File: powerflex/provider/service_datasource_schema.go (hunk 1 above)
Comment (AnikaAgiwal2711) [reply powerflex/provider/service_datasource_schema.go]: Done

File: powerflex/provider/service_datasource_schema.go (hunk 3 above)
Comment (AnikaAgiwal2711) [reply powerflex/provider/service_datasource_schema.go]: Done

Comment (AnikaAgiwal2711) [reply powerflex/provider/service_datasource_schema.go]: Please check it now

File: powerflex/models/service.go (hunk 4 above)
Comment (AnikaAgiwal2711) [reply powerflex/models/service.go]: Done

File: powerflex/provider/service_datasource.go (hunk 5 above)
Comment (AnikaAgiwal2711) [reply powerflex/provider/service_datasource.go]: Replaced

**Review by shenda1 — APPROVED** [review]

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Pull request merged**